url1d5e9d990f_Telsis_jOCP_tgz-pJ8-com_telsis_jocp_sampleApp_BasicOCPAppJ8/src/com/telsis/jocp/OCPLinkManager.java

```diff
@@ -0,0 +1,2057 @@
+                    && managementTidHandler != null) {
+                // The message is for the management task ID and we have a
+                // registered handler.
+                try {
+                    managementTidHandler.queueMessage(message, this);
+                } catch (Exception e) {
+                    log.warn("Exception occurred in management handler "
+                            + managementTidHandler, e);
+                }
+            } else {
+                // We don't know about this task ID. Reject the message.
+                log.warn("Message received for unregistered task: command "
+                        + message.getCommandCode() + ", destination " + taskID);
+                if (!(message instanceof Abort)) { // Do not reply to an Abort
+                    Abort abort = new Abort();
+                    abort.setDestTID(message.getOrigTID());
+                    abort.setOrigTID(message.getDestTID());
+                    abort.setInvalidLegID(Abort.TASK_NOT_RUNNING);
+                    this.queueMessage(abort, null);
+                }
+            }
+        } else {
+            // Unknown message type (but one that OCPMessage knows about!)
+            log.warn("Received an OCP message with an unexpected command type");
+        }
+    }
+
+    /**
+     * Handle incoming link messages. The following link messages are supported:
+     * <p/>
+     * <table>
+     * <tr>
+     * <th>Message type</th>
+     * <th>Action</th>
+     * </tr>
+     * <tr>
+     * <td>{@link Heartbeat}</td>
+     * <td>No action is taken</td>
+     * </tr>
+     * <tr>
+     * <td>{@link StatusRequest}</td>
+     * <td>The link enabled/disabled state, unit ID, cluster ID, and link status
+     * is updated according to the fields in the Status Request. All registered
+     * {@link OCPLinkStateHandler} instances are notified of any changes. A
+     * {@link StatusResponse} is then sent containing the local unit name and
+     * the number of calls in process (determined by the number of registered
+     * task ID handlers).</td>
+     * </tr>
+     * <tr>
+     * <td>{@link LinkCommandUnsupported}</td>
+     * <td>The message details are logged.</td>
+     * </tr>
+     * <tr>
+     * <td>{@link CallGap}</td>
+     * <td>All registered {@link OCPLinkStateHandler} instances are notified of
+     * the call gapping information.</td>
+     * </tr>
+     * <tr>
+     * <td>{@link PreferredUnit}</td>
+     * <td>All registered {@link OCPLinkStateHandler} instances are notified of
+     * the preferred unit.</td>
+     * </tr>
+     * </table>
+     * <p/>
+     * Any unsupported messages results in an
+     * {@link UnknownLinkMessageException} being thrown.
+     *
+     * @param message
+     *            the OCP message to handle
+     * @throws UnknownLinkMessageException
+     *             if the command code is not recognised
+     */
+    private void handleMessage(final LegacyOCPMessage message)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.